### PR TITLE
Add "Hardware setup for testing Qubes OS via openQA" post

### PR DESCRIPTION
**Pull request by SergiiDmytruk**

### CODE REVIEW

**Review by krystian-hebel — COMMENTED** [review]
Few minor issues. I can't say anything about technical correctness of this post as this is first time I got to know openQA in such detail.

The post is definitely one of the longer ones that we have (if not the longest), but I don't know how/if it could be logically split. While I enjoyed reading it, it may be too much for casual readers. Maybe we should have a separate, short (3-4 paragraphs, really) introductory post to what openQA is, where we had used it, a sentence or two about results and link to this one with all the gory details?

**Comment by pietrushnic** [comment]
@marmarek I guess it could be interesting to you or those who handle OpenQA in Qubes OS.

**Comment by marmarek** [comment]
As for checking power state, PiKVM supports monitoring power LED state, and I can confirm it works with MSI - should be more reliable than looking for video stream.

Otherwise looks great :) 

BTW, I still want to use ustreamer with openqa, but as you mentioned, it requires a bit of development. Might happen at some point.

File: blog/content/post/2023-11-11-qubesos-hw-testing.md [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+           |cBLU     |    |cYEL       |  | :            : | |cBLU     |
+           |         |    |           |<-+-:USB Tablet  :-+-+         |
+        AC |         | AC |  System   |  | :            : | |         |
+Mains *----* Sonoff  *----*  Under    |  +-:USB Storage :-+ |  PiKVM  |
+power      |         |    |  Test     |    +============+   |         |
+           |         |    |           +-------------------- >|         |
+           |         |    |           |     HDMI output     |         |
+           +---------+    +-+---------+                     ++--------+
+                 ^          |    ^  ^                        |       ^
+         Power   |  Serial  |    |  |                        |       |
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2023-11-11-qubesos-hw-testing.md]: 9829d5c6722987bf6e17490c78e2007abb856a59

File: blog/content/post/2023-11-11-qubesos-hw-testing.md [hunk 2]
```diff
@@ -0,0 +1,982 @@
...
+The verb "flash" should be interpreted loosely here.  It can be any kind of
+data writing operation meant to prepare SUT for the test.  In this setup it
+means:
+
+1. Setup of USB OTG to provide input and storage via `gadget-control` script
+  (done on PiKVM)
+2. Flashing of BIOS (done on RTE)
+3. Testing that `gadget-control` script is listening for commands
+
+Input is setup here only because `gadget-control` manages both input and
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2023-11-11-qubesos-hw-testing.md]: 3e7f8542b4715bae28e7484535f2542a32fb1b95

File: blog/content/post/2023-11-11-qubesos-hw-testing.md [hunk 3]
```diff
@@ -0,0 +1,982 @@
...
+Setting DV timings
+BT timings set
+DUT is off
+Switched DUT's state from 'unknown'.
+[2023-10-29T20:59:50.930422+01:00] [debug] [pid:26068] Closing SSH connection with 192.168.sut.ip
+[2023-10-29T20:59:50.931789+01:00] [debug] [pid:26068] Passing remaining frames to the video encoder
+{{ skipped uninteresting epilog }}
+```
+
+Tests have finished and SUT is being shutdown.  There is "Device or resource
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2023-11-11-qubesos-hw-testing.md]: 3e7f8542b4715bae28e7484535f2542a32fb1b95

File: blog/content/post/2023-11-11-qubesos-hw-testing.md [hunk 4]
```diff
@@ -0,0 +1,982 @@
...
+```
+
+See respective sections below for more information.
+
+### sol script for openQA worker
+
+This is just another application of `socat`, this time without SSH:
+
+```bash
+socat -U stdio "TCP:$RTE:13541"
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2023-11-11-qubesos-hw-testing.md]: b4f05d56de4a8a5e4a830d21b365a2c1f6093861

**Review by SergiiDmytruk — COMMENTED** [review]
@krystian-hebel
> The post is definitely one of the longer ones that we have (if not the longest), but I don't know how/if it could be logically split. While I enjoyed reading it, it may be too much for casual readers. Maybe we should have a separate, short (3-4 paragraphs, really) introductory post to what openQA is, where we had used it, a sentence or two about results and link to this one with all the gory details?

The idea was to cover how testing setup works rather than to just report that it was used, hence the length.  I meant to organize the post in a way that allows a reader to drop it once he feels overwhelmed with details (each section adds more specific details), maybe a good introduction is missing for this to work as intended.

@marmarek 
> As for checking power state, PiKVM supports monitoring power LED state, and I can confirm it works with MSI - should be more reliable than looking for video stream.

Thanks, that is indeed better.  Mentioned it in ea4ed452216ad07e8ea27876700c6410f2618fad.

> BTW, I still want to use ustreamer with openqa, but as you mentioned, it requires a bit of development. Might happen at some point.

It's possible that extending `os-autoinst` to support PiKVM's VNC would have taken less time than trying out other ways for making remote video work, so it should at least be worth a try.

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**